### PR TITLE
Implement Oxidized "config search" as an API

### PR DESCRIPTION
**Pull request by theochita**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -73,6 +73,7 @@
         Route::put('rules', 'LegacyApiController@add_edit_rule')->name('edit_rule');
         Route::delete('rules/{id}', 'LegacyApiController@delete_rule')->name('delete_rule');
         Route::post('services/{hostname}', 'LegacyApiController@add_service_for_host')->name('add_service_for_host');
+        Route::get('oxidized/configsearch/{searchstring}', 'LegacyApiController@search_oxidized')->name('search_oxidized');
```
Comment (Jellyfrog) [inline routes/api.php]: Any reason why you added this under the admin middleware?  
As far as I know you don't need admin in the web interface for oxidized

Comment (theochita) [reply routes/api.php]: Thanks for replying :)

The reason behind it was that getting to search through config files is something i would imagine only admins should have access to.

I can move it anywhere you want it to be :)

Comment (murrant) [reply routes/api.php]: You do need to be an admin.  Many want it able to be separated, but that isn't an option yet.

Comment (laf) [inline routes/api.php]: Maybe change this to `/config/search/{searchstring}`? We can then use `/config` to return the devices config from the api at a later stage.

Comment (theochita) [reply routes/api.php]: Just updated it to follow your example.

Comment (theochita) [reply routes/api.php]: Thanks for updating the doc @laf  stupid of me to miss that :(

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-57-release-changelog-october-2019/9969/1

**Pull request merged**